### PR TITLE
gdk-pixbuf, gst-plugins-*: migrate `jpeg` to `jpeg-turbo`

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/gtk4.rb [hunk 1]
```diff
@@ -31,7 +33,10 @@ class Gtk4 < Formula
   depends_on "glib"
   depends_on "graphene"
   depends_on "hicolor-icon-theme"
+  depends_on "jpeg-turbo"
   depends_on "libepoxy"
+  depends_on "libpng"
+  depends_on "libtiff"
```
Comment (cho-m) [inline Formula/gtk4.rb:39]: Mandatory dependencies as of 4.5.0 https://gitlab.gnome.org/GNOME/gtk/-/blob/4.6.6/NEWS#L496. Used for some features.

**Comment by cho-m** [comment]
Linux build failures can be ignored since they are due to dropping `mesa` bottle.

So, `gst-plugins-good` , `gtk4`, and `libgxps` will lose Linux bottles, which is expected.

This will then cascade into vast majority of dependents failing. Just need to check for any `jpeg` linkage failures in dependents that don't fail on `brew fetch`/`brew install --only-dependencies`.

**Comment by cho-m** [comment]
129 dependents failed due to dropping Linux bottles. I didn't see anything specifically related to this PR as there are 258 failed steps with each dependent failing once on `brew install --only-dependencies` and again on `brew install`.

This might get a bit tedious to review PRs for formula with large dependency trees (`icu4c`, `libxml2`, ...) since we are going to have to sift through hundreds of failures to find the real ones.

**Comment by carlocab** [comment]
> This might get a bit tedious to review PRs for formula with large dependency trees (`icu4c`, `libxml2`, ...) since we are going to have to sift through hundreds of failures to find the real ones.

Would be good to fix this in `test-bot`.

**Pull request closed without merging**